miner/worker.go

```diff
@@ -92,7 +92,7 @@ type Result struct {
 type worker struct {
 	mu sync.Mutex
 
-	agents []Agent
+	agents map[Agent]struct{}
```

Comment:
Aside from the fact that changing this data type resulted in a discussion which now caused you to write the very function you wanted to avoid writing by making this change, you don't have to write this function to delete an agent.

Rather, you can make use of `agent.index` to know which element in the slice to delete. Each agent has it's index set on creation: https://github.com/ethereum/go-ethereum/blob/master/miner/miner.go#L115

This will shorten the delete operation to `a = append(a[:i], a[i+1:]...)`